### PR TITLE
fix_shortest_path_crash

### PR DESCRIPTION
**Pull request by nevermore3**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Description:

Graph crash when running tck manually 
image
Can't be reproduced consistently.
Tck is run with 16 concurrent threads.

## How do you solve it?

 shared variables are locked before use in multi-threading environment

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/executor/algo/ShortestPathBase.h [hunk 1]
```diff
@@ -86,6 +80,7 @@ class ShortestPathBase {
   std::unordered_map<std::string, std::string>* stats_{nullptr};
   size_t maxStep_;
   bool singleShortest_{true};
+  folly::SpinLock statsLock_;
```
Comment (Shylock-Hg) [inline src/graph/executor/algo/ShortestPathBase.h:83]: How about other memebers?

Comment (nevermore3) [reply src/graph/executor/algo/ShortestPathBase.h:83]: other members will not be written by multiple threads at the same time

File: src/graph/executor/algo/SingleShortestPath.cpp [hunk 2]
```diff
@@ -264,6 +264,9 @@ folly::Future<bool> SingleShortestPath::buildEvenPath(size_t rowNum,
       return false;
     }
     for (auto& meetVertex : vertices) {
+      if (!meetVertex.isVertex()) {
```
Comment (Shylock-Hg) [inline src/graph/executor/algo/SingleShortestPath.cpp:267]: Why this isn't vertex?

**Review by Aiee — APPROVED** [review]

File: src/graph/executor/algo/SingleShortestPath.cpp (hunk 2 above)
Comment (nevermore3) [reply src/graph/executor/algo/SingleShortestPath.cpp:267]: vertices may contain EMPTY values ​​in exceptional cases

**Review by Shylock-Hg — APPROVED** [review]

**Pull request merged**